### ISSUE TITLE
Slow nightly load

### ISSUE THREAD

**Issue by kimrutherford** [pombase/pombase-chado#1118]
Adding `PMID_37787768_phaf.tsv` made the nightly load much slower.  It still finishes before the morning but I think there must be something wrong in the PHAF import code.

I made quite a few small changes to improve performance last week but it wasn't enough.

**Comment by kimrutherford** [comment]
I did some profiling of the PHAF loading to find why it's so slow.  I think I've improved things by:

  - adding a new index in Chado to speed up searching for existing features (genes, alleles and genotypes for PHAF loading)
  - adding some more caching in the genotype/allele creation code so that fewer Postgres queries are needed

On my desktop these two changes make PHAF loading ~4 times faster which is a much better result than I expected.  Hopefully the speedup on oliver1 will be similar.  I'll see how things go on Thursday night.
